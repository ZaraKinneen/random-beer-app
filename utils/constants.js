export const CORS = "https://cors-anywhere.herokuapp.com";
export const BASE_URL = "http://api.brewerydb.com/v2";
export const API_KEY = "a5c1b917e7ba62dcd79f434ed73bc72d";

export const IMAGE_PLACEHOLDER =
  "https://touch.daft.ie/static/images/fallbacks/no-image-size740x480.jpg";
