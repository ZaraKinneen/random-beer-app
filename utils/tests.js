import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

const theme = {
  colors: {
    primary: "#355C7D",
    secondary: "#C06C84",
  },
  fonts: {
    light: "#FFFFFF",
    dark: "#000000",
    highlight: "#FFCC00",
  },
};

const ThemeWrapper = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
