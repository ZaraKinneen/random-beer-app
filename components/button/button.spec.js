import { render } from "../../utils/tests";
import ButtonComponent from "./button";

describe("ButtonComponent", () => {

  it("should match snapshot", () => {
    const { container } = render(<ButtonComponent />);

    expect(container).toMatchSnapshot();
  });
});
