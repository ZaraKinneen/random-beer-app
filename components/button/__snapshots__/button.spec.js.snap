// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should match snapshot 1`] = `
<div>
  <div
    class="button__ButtonContainer-banjw0-0 iAsILJ"
  >
    <button
      class="button__Button-banjw0-2 kZkJPj"
    >
      <span
        class="button__ButtonDetail-banjw0-3 esgwoO"
      >
        <p
          class="button__Text-banjw0-4 elNeGS"
        />
        <img
          class="button__Icon-banjw0-1 fnTJtB"
        />
      </span>
    </button>
  </div>
</div>
`;
